### PR TITLE
module/tls: Make TLS validity period configurable

### PR DESCRIPTION
**Pull request by kalmog**
Fixes #INST-239

### CODE REVIEW

File: config.tf [hunk 1]
```diff
@@ -391,6 +391,16 @@ This field is mandatory if `tectonic_ca_cert` is set.
 EOF
 }
 
+variable "tectonic_tls_validity_period" {
+  description =<<EOF
+Validity period of the self-signed certificates (in hours).
+Default is 3 years.
```
Comment (s-urbaniak) [inline config.tf]: just a nit: let's add that "This setting is ignored if user provided certificates are used."

Comment (kalmog) [reply config.tf]: Done :)

**Comment by kalmog** [comment]
test this please

**Comment by kalmog** [comment]
test this please

**Comment by kalmog** [comment]
test this please

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
retest this please

**Comment by cpanato** [comment]
tests are green, just did not report correctly to GitHub, this PR can be merged

**Pull request merged**